go2eiko.py

```diff
@@ -0,0 +1,439 @@
+
+
+# Constants
+EXTENSIONS = ['png', 'jpg', 'jpeg']
+AUTH = None  # TODO: might not be the most secured way to store the credentials
+TEMP_FOLDER = "temp"
+
+
+# Informations about the repository
+OWNER = "frzyc"
+REPO = "genshin-optimizer"
+
+
+# User specific
+MASTER_OUTPUT_PATH = None
+FORCE = None
+METHOD = None
+
+
+def sparse_checkout(path_to_checkout: str, path_to_store: str = TEMP_FOLDER):
+
+    # Get current path
+    current_path = os.getcwd()
+
+    # Clone the repository
+    os.makedirs(path_to_store, exist_ok=True)
+    os.chdir(path_to_store)
+    os.system(f"git clone --filter=blob:none --no-checkout https://github.com/frzyc/genshin-optimizer.git")
+    os.chdir("genshin-optimizer")
+    os.system(f"git config core.sparseCheckout true")
+    os.system(f"echo {path_to_checkout} >> .git/info/sparse-checkout")
+    os.system("git checkout")
+    os.system("git pull")
+
+    # Go back to the original folder
+    os.chdir(current_path)
+
+
+def intex_dot_ts2name_converter(index_dot_ts: str, old_filename: str) -> str:
+    
+    # Convert the name using the index.ts file
+    new_filename = old_filename.split(".")[0]
+    ext = old_filename.split(".")[1]
+    pattern = r"import ([A-Za-z0-9]+) from '.\/([A-Za-z0-9_]+)\.([a-z]+)'"
+    for line in index_dot_ts.split("\n"):
+        if re.match(pattern, line):
+            new_filename = re.search(pattern, line).group(1)
+            ext = re.search(pattern, line).group(3)
+            if re.search(pattern, line).group(2) + "." + ext == old_filename:
+                break
+    
+    # Convert the resulting name to match eikonomiya's convention
+    d = {
+        "constellation1": "c1",
+        "constellation2": "c2",
+        "constellation3": "c3",
+        "constellation4": "c4",
+        "constellation5": "c5",
+        "constellation6": "c6",
+        "icon": "face",
+        "banner": "namecard",
+    }
+    return f"{d[new_filename]}.{ext}" if new_filename in d else old_filename
+
+
+def download_folder(
+    path: str,
+    output_path: Union[str, callable],
+    name_converter: callable,
+):
+    """
+    Download all the files in a folder of a GitHub repository.
+    
+    Parameters
+    ----------
+    path : str
+        Path of the folder in the repository. Usually it is right after
+        "tree/master/" in the repository's URL.
+    output_path : Union[str, callable]
+        Path of the folder where the files will be saved. Can be a relative
+        path or an absolute path. If a callable, it should take a single
+        argument corresponding to the name of the file as stored in the
+        repository, and return the path of the file as it should be saved.
+    name_converter : callable
+        The function that converts the name of the file. It should take a
+        single argument corresponding to the name of the file as stored in the
+        repository, and return the name of the file as it should be saved.
+    """
+
+    if METHOD == "checkout":
```

Comment:
is it possible to delete the temp folder after that ? it currently uses 190MB after the operation

Reply:
Working on it. Some files in read only are giving me headache

Reply:
Fixed in c61923c0fa68e435d494832de8b02fe3233665c7
You can now also use the flag `--keep` to avoid the deletion of the `temp` folder